unblob/file_utils.py

```diff
@@ -9,3 +10,38 @@ def snull(content: bytes):
 def round_up(size: int, alignment: int):
     """Round up size to the alignment boundary."""
     return alignment * math.ceil(size / alignment)
+
+
+class LimitedStartReader(io.BufferedIOBase):
+    """Wrapper for open files, which
+    enforces that seekeng earlier than the start offset is not possible.
+    """
+
+    def __init__(self, file: io.BufferedIOBase, start: int):
+        self._file = file
+        self._start = start
+        self._file.seek(start)
+
+    def seek(self, offset: int, whence=io.SEEK_SET):
+        new_pos = self._file.seek(offset, whence)
+        if new_pos < self._start:
+            new_pos = self._file.seek(self._start, io.SEEK_SET)
+        return new_pos
+
+    def write(self, *args, **kwargs):
+        raise TypeError("MUST NOT call write method")
+
+    def detach(self, *args, **kwargs):
```

Comment:
Don't know if it is a good / effective idea, but I would write simple test cases for each of these functions, at least check them that they return with the result of calling the same function (with the same parameters) on the `self._file`